### ISSUE TITLE
`set_cookie` will no longer set path by default

### ISSUE THREAD

**Issue by mgalbier** [pallets/werkzeug#2672]
It seems that since version 2.3.0 of Werkzeug the behaviour of `set_cookie` has changed. The documentation of the method says that by default the cookie will span the whole domain. In older versions this was true: if the parameter `path` was not set it defaulted to `"/"`. Now it defaults to `None`, which leads to `Path` being omitted in the `Set-Cookie` part of the header in the response sent to the client. The browser (tested with Firefox) seems then to just take the path from the request URL.

```
# return from an endpoint like /cookie/set
response.set_cookie('test_cookie', cookie_payload, expires=exp_in_thirty_minutes, secure=True,
                        samesite='Strict')
# --> cookie for client will have path set to /cookie
```

Expected is that `Path` of the cookie will be set to "/" according to the documentation of `set_cookie`.

Environment:

- Python version: 3.10
- Werkzeug version: 2.3.1

**Comment by davidism** [comment]
That's correct. If the path is not given, browsers do not set a limit on the path, therefore it does not need to be sent by default. I can't reproduce your result, Firefox does not use the current path.

**Comment by mgalbier** [comment]
I have here an minimal example:

```
from werkzeug.wrappers import Request, Response


def set_cookie(request):
    response = Response("Cookie set!")
    response.set_cookie("my_cookie", "cookie_value")
    return response


def application(environ, start_response):
    request = Request(environ)
    response = None

    if request.path == '/cookie/set':
        response = set_cookie(request)
    else:
        response = Response("Page not found", status=404)

    return response(environ, start_response)


if __name__ == '__main__':
    from werkzeug.serving import run_simple

    run_simple('localhost', 5000, application)
```

after visiting `http://localhost:5000/cookie/set` I have the following cookie:

![image](https://user-images.githubusercontent.com/17177211/235173132-86c49c0b-e54d-4de1-a5cb-3b398bb05bd8.png)

If this behavior differs it must be some Browser differences. 
But in my understanding it seems to match the behavior specified here [RFC 6265 ](https://datatracker.ietf.org/doc/html/rfc6265#section-5.2.4) using the definition of `default-path` from [here](https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.4).

**Comment by mgalbier** [comment]
OK, that is confusing. I just tried it with Chrome as well (running the minimal example from my last response) and I get the same behavior:

![image](https://user-images.githubusercontent.com/17177211/235176277-63da0d00-2ffe-485d-b6d0-bb81d97dd052.png)

Output from `pip freeze`:

```
MarkupSafe==2.1.2
Werkzeug==2.3.1
```
